core/graphelier-service/db/mongo.go

```diff
@@ -88,3 +95,57 @@ func (c *Connector) GetMessages(instrument string, timestamp uint64, latestFullS
 
 	return results, nil
 }
+
+// GetMessagesWithPagination returns an messages given an instrument, sod_offset with pagination
+func (c *Connector) GetMessagesWithPagination(instrument string, sodOffset int64, paginator *models.Paginator) (results []*models.Message, err error) {
+	collection := c.Database("graphelier-db").Collection("messages")
+	var filterKey string
+	var sortDirection int8
+	if paginator.NMessages < 0 {
+		filterKey = "$lt"
+		sortDirection = -1
+	} else {
+		filterKey = "$gt"
+		sortDirection = 1
+	}
+	filter := bson.D{
+		{Key: "instrument", Value: instrument},
+		{Key: "sod_offset", Value: bson.D{
+			{Key: filterKey, Value: sodOffset},
+		}},
+	}
+
+	findOptions := options.Find()
+	findOptions.SetSort(bson.D{{Key: "instrument", Value: 1}, {Key: "sod_offset", Value: sortDirection}})
+	findOptions.SetLimit(abs(paginator.NMessages))
+
+	cursor, err := collection.Find(context.TODO(), filter, findOptions)
+	if err != nil {
+		return nil, err
+	}
+	defer cursor.Close(context.TODO())
+
+	for cursor.Next(context.TODO()) {
+		var m models.Message
+		err := cursor.Decode(&m)
+		if err != nil {
+			return nil, err
+		}
+
+		results = append(results, &m)
+	}
+
+	if err := cursor.Err(); err != nil {
+		return nil, err
+	}
+
+	return results, nil
+}
+
+// Default Abs func for golang takes a float and returns a float. Need one for ints
+func abs(n int64) int64 {
+	if n < 0 {
+		return -n
+	}
+	return n
+}
```

Comment:
Is ternary not a preferred option in go ?

Reply:
There is no ternary operator in go